scripts/compare-versions/src/main.rs

```diff
@@ -0,0 +1,228 @@
+                .unwrap()
+                .gt(&latest_indexed_version)
+                .then_some(Version::from_str(&r.head_branch[1..]).unwrap())
+        })
+        .collect();
+
+    // In case something went wrong above, we only want a maximum of 3 versions each so that we do not start too many CI jobs.
+    Ok(new_versions[..std::cmp::min(new_versions.len(), MAX_VERSIONS)].to_vec())
+}
+
+fn parse_latest_indexed_version(channel: &Document, package: &str) -> Version {
+    Version::from_str(channel["pkg"][package]["version"].as_str().expect(&format!(
+        "Could not parse {} version str from {} toml",
+        package, channel
+    )))
+    .expect(&format!("Could not create version from {}", package))
+}
+
+fn fmt_versions(forc_version: &str, fuel_core_version: &str) -> String {
+    format!("forc-{}@fuel-core-{}", forc_version, fuel_core_version)
+}
+
+fn print_selected_versions<'a>(
+    forc_versions: &[Version],
+    fuel_core_versions: &[Version],
+) -> String {
+    let mut output = String::new();
+
+    for forc in forc_versions {
+        for fuel_core in fuel_core_versions {
+            let formatted_versions = fmt_versions(&forc.to_string(), &fuel_core.to_string());
+            output.push_str(&formatted_versions);
+            output.push_str("\n");
+        }
+    }
+
+    print!("{}", output);
+    // Return output for testing purposes
+    output.to_string()
+}
+
+fn main() -> Result<()> {
+    let handle = ureq::builder().user_agent("fuelup").build();
+
+    let toml_resp = match handle.get(&CHANNEL_FUEL_LATEST_TOML_URL).call() {
+        Ok(r) => r
+            .into_string()
+            .expect("Could not convert channel to string"),
+        Err(_) => {
+            eprintln!(
+                "Could not download channel-fuel-latest.toml from {}; re-generating channel.",
+                &CHANNEL_FUEL_LATEST_TOML_URL
+            );
+
+            let sway_runs = get_workflow_runs(SWAY_REPO)?;
+            let fuel_core_runs = get_workflow_runs(FUEL_CORE_REPO)?;
+
+            let latest_sway_version =
+                Version::from_str(&sway_runs.workflow_runs[0].head_branch[1..]).unwrap();
+            let latest_fuel_core_version =
+                Version::from_str(&fuel_core_runs.workflow_runs[0].head_branch[1..]).unwrap();
+            print_selected_versions(&[latest_sway_version], &[latest_fuel_core_version]);
+            std::process::exit(0);
+        }
+    };
+
+    let channel_doc = toml_resp
+        .parse::<Document>()
+        .expect("invalid channel.toml parsed");
+
+    let forc_versions = collect_new_versions(&channel_doc, SWAY_REPO).unwrap();
+    let fuel_core_versions = collect_new_versions(&channel_doc, FUEL_CORE_REPO).unwrap();
+
+    select_versions(&channel_doc, forc_versions, fuel_core_versions);
+    Ok(())
+}
+
+fn select_versions(
+    channel: &Document,
+    mut forc_versions: Vec<Version>,
+    mut fuel_core_versions: Vec<Version>,
+) -> String {
```

Comment:
This function actually performs the I/O and prints the main output of the program, but it's a little unclear looking at the signature. Perhaps rather than calling `print_selected_versions` within this function, we could instead move that into a step *after* `select_versions` is called?